assignment_1/1_permutations.py

```diff
@@ -0,0 +1,44 @@
+#Given two strings, write a method to decide if one is a permutation of the other
+# Magdalena Grodzińska
+
+import re
+from collections import Counter
+
+# helper funtions
+
+def validate_input(input):
+    pattern = '[a-zA-Z]+'
+
+    result = re.fullmatch(pattern, input)
+
+    if result == None:
+        return False
+    else:
+        return True
+
+def check_letters_match(text1, text2):
+    res1 = Counter(text1)
+    res2 = Counter(text2)
+
+    return res1 == res2
+
+# main function
+
+def check_permutation(text1, text2):
+    if validate_input(text1) and validate_input(text2):
+        return check_letters_match(text1, text2)
+    else:
+        return False
+
+# tests
```

Comment:
It might be convenient to just formulate these as real unit tests:

https://docs.python.org/3.6/library/unittest.html#basic-example